sema/subyt/sources.py

```
@@ -310,7 +310,15 @@ def __enter__(self):
                 xml_str = xmlfile.read()
                 xdict = xmlasdict.parse(xml_str)
                 # unpack root wrappers
-                data = xdict.unpack()
+                try:
+                    xdict = xmlasdict.parse(xml_str)
+                    # unpack root wrappers
+                    data = xdict.unpack()  # type: ignore
+                except Exception as e:
+                    log.error(
+                        f"Failed to parse XML file {self._xml}: {str(e)}"
+                    )
```

Ruff

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)